### PR TITLE
replace defer with using unLock directly 

### PR DESCRIPTION
**Pull request by zouyee**
Signed-off-by: Zou Nengren <zouyee1989@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
using unLock directly replace defer

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/v1alpha1/waiting_pods_map.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"sync"
 	"time"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (denkensk) [inline pkg/scheduler/framework/v1alpha1/waiting_pods_map.go]: No need to change here?

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 2]
```diff
@@ -399,7 +399,6 @@ func (p *PriorityQueue) flushBackoffQCompleted() {
 // to activeQ.
 func (p *PriorityQueue) flushUnschedulableQLeftover() {
 	p.lock.Lock()
-	defer p.lock.Unlock()
```
Comment (draveness) [inline pkg/scheduler/internal/queue/scheduling_queue.go]: I'd prefer to keep this one

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 3]
```diff
@@ -545,14 +545,14 @@ func (p *PriorityQueue) AssignedPodUpdated(pod *v1.Pod) {
 // queue and the head is the highest priority pod.
 func (p *PriorityQueue) MoveAllToActiveOrBackoffQueue(event string) {
 	p.lock.Lock()
-	defer p.lock.Unlock()
```
Comment (draveness) [inline pkg/scheduler/internal/queue/scheduling_queue.go]: and this one

File: pkg/scheduler/algorithm/predicates/metadata.go [hunk 4]
```diff
@@ -787,13 +787,13 @@ func getTPMapMatchingIncomingAffinityAntiAffinity(pod *v1.Pod, allNodes []*sched
 	topologyPairsAntiAffinityPodsMaps = newTopologyPairsMaps()
 	appendResult := func(nodeName string, nodeTopologyPairsAffinityPodsMaps, nodeTopologyPairsAntiAffinityPodsMaps *topologyPairsMaps) {
 		lock.Lock()
```
Comment (draveness) [inline pkg/scheduler/algorithm/predicates/metadata.go]: this one

**Review by draveness — CHANGES_REQUESTED** [review]
Generally looks good, a few nits. 

I think it would be great to use the defer when the function has multiple lines (maybe more than 5?), the Golang team has been optimizing the cost of defer for quite a while, it's not that bad now. We could choose the code readability and avoid mistakes instead of the tiny performance improvement in some functions.

**Comment by denkensk** [comment]
@zouyee  Thanks~ Can you add Why we need to replace defer with using unLock directly?

File: pkg/scheduler/framework/v1alpha1/waiting_pods_map.go (hunk 1 above)
Comment (draveness) [reply pkg/scheduler/framework/v1alpha1/waiting_pods_map.go]: This won't be a problem IMO. This was automatically introduced by goimports, and it is not worth to fight the tools.

**Comment by ahg-g** [comment]
Thanks for your PR, but there is no clear benefit from doing this.

If you want to contribute a PR, please make sure you have a clear explanation why we need the PR and what issue it solves.

/close

**Pull request closed without merging**